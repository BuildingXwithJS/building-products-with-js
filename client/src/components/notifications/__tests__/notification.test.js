// npm packages
import configureMockStore from 'redux-mock-store';

// our packages
import NotificationWrapper, {Notification} from '../notification';

// create mock store
const mockStore = configureMockStore();

const notification = {
  id: '0',
  text: 'Test notification',
  alertType: 'test',
};

test('# NotificationWrapper', () => {
  const store = mockStore({});
  const wrapper = shallow(<NotificationWrapper notification={notification} store={store} />);
  expect(wrapper).toMatchSnapshot();
});

test('# Notification', () => {
  const onClick = (id) => expect(id).toBe(notification.id);
  const component = <Notification notification={notification} onRemoveNotificationClick={onClick} />;
  const wrapper = shallow(component);
  expect(wrapper).toMatchSnapshot();
  // test interaction
  const app = mount(component);
  const item = app.find('button');
  item.simulate('click');
});

