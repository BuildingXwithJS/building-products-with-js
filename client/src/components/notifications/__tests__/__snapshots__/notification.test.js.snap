// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# Notification 1`] = `
<div
  className="alert alert-dismissible alert-test"
  role="alert"
>
  <button
    aria-label="Close"
    className="close"
    data-dismiss="alert"
    onClick={[Function]}
    type="button"
  >
    <span
      aria-hidden="true"
    >
      Ã—
    </span>
  </button>
  Test notification
</div>
`;

exports[`# NotificationWrapper 1`] = `
<Notification
  notification={
    Object {
      "alertType": "test",
      "id": "0",
      "text": "Test notification",
    }
  }
  onRemoveNotificationClick={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
